File: Programacion/S2/Unidad 2/src/pb_05_032019_AppPila/Pila.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pb_05_032019_AppPila;

/**
 *
 * @author aa613
 */
public class Pila {
    private double [] datos;
    private int tope;

    public Pila() {
        datos = new double [10];
        tope =-1; // vacia 
    }
    public Pila(int n){
        if(n<1)
            n = 10;
        
        datos = new double [n];
        tope = -1; // pila base 
    }
    public int getCapacidad(){
        return datos.length;
    }
    public boolean estaVacia(){
        
//        if(tope == -1)
//            return true;
//        else 
//            return false;
         return tope == -1;
    }
    public boolean estaLlena(){
        return tope == datos.length-1;
    }
    public void poner(double x){
        if(!estaLlena()){
            tope++;
            datos[tope] =x;
        }
    }
    public double quitar(){
        if(!estaVacia()){
            tope--;
            return datos[tope+1];
        }
        else 
            return 0;
    }

    @Override
    public String toString() {
       String cad = "[";
       for(int i =0;i<=tope;i++){
           cad+=datos[i];
           
           if(i<tope)
               cad+= ",";
       }
       cad+= "]";
       return cad;
    }
    
    
    
    
}

File: Programacion/S4/appPilaEstatica/Pila.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Pila.cpp
 * Author: The Survivor
 * 
 * Created on 30 de marzo de 2020, 04:27 PM
 */
#include <iostream>
#include "Pila.h"

Pila::Pila()
{
     tope = -1;
     for(int i = 0; i < tam; i++)
        datos[i] = 0;//inicializa a ceros el arreglo
     //obligatorio inicializar en c++
}

bool Pila::push(int valor)
{
    if(tope == tam -1)
        return false;
    else
    {
        datos[++tope]= valor;
        return true;
    }
}

//pase de la referencia
bool Pila::pop(int &valor)
{
     if(tope == -1)
        return false;
     else
     {
        valor = datos[tope--];
        return true;
     }
}

void Pila::mostrar()
{
    cout<<"\nDatos en la pila : \n";
    for(int i = 0; i <= tope; i++)
        cout<<"\n Dato[ "<<i<<" ] -> "<<datos[i];
}

